applications/fishing-map/src/features/map-features/map-features.hooks.ts

```diff
@@ -0,0 +1,136 @@
+import { useSelector, useDispatch } from 'react-redux'
+import GFWAPI from '@globalfishingwatch/api-client'
+import {
+  ExtendedFeature,
+  ExtendedFeatureVessel,
+  InteractionEvent,
+} from '@globalfishingwatch/react-hooks'
+import { resolveEndpoint, Dataview } from '@globalfishingwatch/dataviews-client'
+import { Generators } from '@globalfishingwatch/layer-composer'
+import {
+  selectWorkspaceDataviews,
+  selectTemporalgridDataviews,
+} from 'features/workspace/workspace.selectors'
+import { selectTimerange } from 'routes/routes.selectors'
+import { setClickedEvent, setFeatureVessels, selectClickedEvent } from './map-features.slice'
+
+export const useClickedEventConnect = () => {
+  const dispatch = useDispatch()
+  const clickedEvent = useSelector(selectClickedEvent)
+
+  const dataviews = useSelector(selectWorkspaceDataviews)
+  const temporalgridDataviews = useSelector(selectTemporalgridDataviews)
+  const { start, end } = useSelector(selectTimerange)
+
+  const dispatchClickedEvent = (event: InteractionEvent | null) => {
+    if (event === null) {
+      dispatch(setClickedEvent(null))
+    }
+    if (!event || !event.features) return
+    dispatch(setClickedEvent(event))
+    // TODO should work for multiple features
+    const feature: ExtendedFeature = event.features[0]
+    if (!dataviews || !feature || !feature.temporalgrid) return
+
+    // TODO We assume here that temporalgrid dataviews appear in the same order as sublayers are set in the generator, ie indices will match feature.temporalgrid.sublayerIndex
+    const dataview = temporalgridDataviews[feature.temporalgrid.sublayerIndex]
+    // TODO How to get the proper id? Should be fishing_v4
```

Comment:
good catch, we'll need to think deeply about this as I don't have an answer right now